### DISCUSSION TITLE
[ui] combobox: how to redirect to an URL when an item is selected?

### DISCUSSION THREAD

**Discussion by gpluess** [alpinejs/alpine#4221] [1. Help]
How can I redirect a user when they select an option in the combobox?

The below code works for the click event, but not for the keyboard event.

```html
<li x-combobox:option :value="company"
    @click="selectCompany(company)"
    @keydown.enter="selectCompany(company)">
    <!-- more stuff -->
</li>
```

And somehow I have the feeling that this is not how it should be solved. Is there a better way?

**Comment by ekwoka** [comment]
window.location = newurl

but then shouldn't those be buttons? or just links?

**Reply by gpluess** [reply]
The actual redirect is not the issue. My problem is, that it's not working when selecting an item by keyboard using "enter". I assume the component itself is already registering a keyboard event listener that prevents mine to work correctly.

**Reply by ekwoka** [reply]
You can see here https://github.com/alpinejs/alpine/blob/main/packages/ui/src/combobox.js



That it doesn't. More likely, the focus is just not actually inside that element because an `li` won't otherwise be focusable.


So while you think you are pressing enter on that li, you are actually pressing enter somewhere else.
